pkg/model/provider/bedrock/client.go

```diff
@@ -0,0 +1,397 @@
+
+	if c.config.Temperature > 0 {
+		request["temperature"] = c.config.Temperature
+	}
+
+	if c.config.TopP > 0 {
+		request["top_p"] = c.config.TopP
+	}
+
+	// Add tools if provided
+	if len(requestTools) > 0 {
+		claudeTools := []map[string]any{}
+		for _, tool := range requestTools {
+			inputSchema, err := tools.SchemaToMap(tool.Parameters)
+			if err != nil {
+				return nil, fmt.Errorf("failed to convert tool parameters: %w", err)
+			}
+			claudeTools = append(claudeTools, map[string]any{
+				"name":         tool.Name,
+				"description":  tool.Description,
+				"input_schema": inputSchema,
+			})
+		}
+		request["tools"] = claudeTools
+	}
+
+	return json.Marshal(request)
+}
+
+// buildTitanRequest builds request for Amazon Titan models
+func (c *Client) buildTitanRequest(messages []chat.Message) ([]byte, error) {
+	// Titan uses a simpler format - concatenate all messages
+	var prompt strings.Builder
+	for _, msg := range messages {
+		if msg.Content != "" {
+			prompt.WriteString(msg.Content)
+			prompt.WriteString("\n")
+		}
+	}
+
+	request := map[string]any{
+		"inputText": strings.TrimSpace(prompt.String()),
+		"textGenerationConfig": map[string]any{
+			"maxTokenCount": c.config.MaxTokens,
+			"temperature":   c.config.Temperature,
+			"topP":          c.config.TopP,
+		},
+	}
+
+	return json.Marshal(request)
+}
+
+// buildLlamaMistralRequest builds request for Meta Llama and Mistral models
+func (c *Client) buildLlamaMistralRequest(messages []chat.Message) ([]byte, error) {
+	// Llama/Mistral use instruction format
+	var prompt strings.Builder
+	for _, msg := range messages {
+		if msg.Role == chat.MessageRoleSystem {
+			prompt.WriteString("<<SYS>>\n")
+			prompt.WriteString(msg.Content)
+			prompt.WriteString("\n<</SYS>>\n\n")
+		} else if msg.Role == chat.MessageRoleUser {
+			prompt.WriteString("[INST] ")
+			prompt.WriteString(msg.Content)
+			prompt.WriteString(" [/INST]\n")
+		} else if msg.Role == chat.MessageRoleAssistant {
+			prompt.WriteString(msg.Content)
+			prompt.WriteString("\n")
+		}
+	}
+
+	request := map[string]any{
+		"prompt": strings.TrimSpace(prompt.String()),
+	}
+
+	if c.config.MaxTokens > 0 {
+		request["max_gen_len"] = c.config.MaxTokens
+	}
+
+	if c.config.Temperature > 0 {
+		request["temperature"] = c.config.Temperature
+	}
+
+	if c.config.TopP > 0 {
+		request["top_p"] = c.config.TopP
+	}
+
+	return json.Marshal(request)
+}
+
+// getModelFamily determines the model family from the model ID
+func getModelFamily(modelID string) string {
+	lower := strings.ToLower(modelID)
+	switch {
+	case strings.Contains(lower, "anthropic.claude"):
+		return "anthropic"
+	case strings.Contains(lower, "amazon.titan"):
+		return "titan"
+	case strings.Contains(lower, "meta.llama"):
+		return "llama"
+	case strings.Contains(lower, "mistral"):
+		return "mistral"
+	default:
+		return "unknown"
+	}
+}
+
+// ID returns the model provider ID
+func (c *Client) ID() string {
+	return c.config.Provider + "/" + c.config.Model
```

Comment:
We use this to track cost, this would return something like `bedrock/....` but over at models.dev the provider id is `amazon-bedrock` so we should return "amazon-bedrock/"+c.config.Model

Reply:
So which one would be the preferred format?

Reply:
Let's call it `amazon-bedrock`, it's what they call it here https://aws.amazon.com/bedrock/